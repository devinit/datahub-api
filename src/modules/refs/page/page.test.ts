import 'jest';
import {
    getCountryProfilePageData,
    getPovertyBubbleChartPageData,
    getOdaDonorBubbleChartPageData,
    getUnbundlingOdaPageData,
    getUnbundlingOOfPageData,
    getWhereThePoorPageData,
    getAboutPageData,
    getFrontPageData,
    getSpotlightPageData,
    getUnbundlingAidPageData,
    getBubbleChartAnnotationPageData,
    getWhoAreTheGlobalP20PageData,
    getProfileHeaderPageData
} from '.';
import * as prettyFormat from 'pretty-format';

describe('Page data', () => {
    it.skip('should return page data for a country profile', async () => {
        const pageData = await getCountryProfilePageData();
        expect(pageData.length).toBeGreaterThan(2);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 100000);
    it.skip('should return page data for the Poverty bubble chart', async () => {
        const pageData = await getPovertyBubbleChartPageData();
        //  console.log(pageData[0]);
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for the ODA donor bubble chart', async () => {
        const pageData = await getOdaDonorBubbleChartPageData();
        //  console.log(pageData[0]);
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for the Unbundling Oda', async () => {
        const pageData = await getUnbundlingOdaPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for  Unbundling OOf', async () => {
        const pageData = await getUnbundlingOOfPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for Where The Poor', async () => {
        const pageData = await getWhereThePoorPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for about', async () => {
        const pageData = await getAboutPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for front page', async () => {
        const pageData = await getFrontPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for spotlight page', async () => {
        const pageData = await getSpotlightPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for unbundling-aid page', async () => {
        const pageData = await getUnbundlingAidPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for bubblechartAnnotation page', async () => {
        const pageData = await getBubbleChartAnnotationPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it.skip('should return page data for whoAreTheGlobalP20 page', async () => {
        const pageData = await getWhoAreTheGlobalP20PageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
    it('should return page data for profileHeader page', async () => {
        const pageData = await getProfileHeaderPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
});
