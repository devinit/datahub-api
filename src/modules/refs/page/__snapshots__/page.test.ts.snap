// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page data should return page data for  Unbundling OOf 1`] = `
"Array [
  Object {
    \\"id\\": \\"Unbundling aid\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Explore and compare funding priorities for other official flows (OOFs)\\",
    \\"title\\": \\"Unbundling aid\\",
  },
]"
`;

exports[`Page data should return page data for Where The Poor 1`] = `
"Array [
  Object {
    \\"id\\": \\"Unbundling aid\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Explore and compare funding priorities for other official flows (OOFs)\\",
    \\"title\\": \\"Unbundling aid\\",
  },
]"
`;

exports[`Page data should return page data for Where The Poor 2`] = `
"Array [
  Object {
    \\"id\\": \\"poverty-visualisation\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Between 1990 and 2010, extreme poverty was cut in half and Explore plausible poverty outcomes for 2030 based on models of consumption\\",
    \\"title\\": \\"Where are the poor and where will they be?\\",
  },
  Object {
    \\"id\\": \\"Source\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Development Initiatives based on OECD DAC data.Note that figures are rounded: precise data are available for download on the methodology page\\",
    \\"title\\": \\"Source\\",
  },
  Object {
    \\"id\\": \\"download-the-data\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"For more information, see the DI data hub methodology page\\",
    \\"title\\": \\"Download the data\\",
  },
  Object {
    \\"id\\": \\"poverty-visualisation\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Between 1990 and 2010, extreme poverty was cut in half and Explore plausible poverty outcomes for 2030 based on models of consumption\\",
    \\"title\\": \\"Where are the poor and where will they be?\\",
  },
  Object {
    \\"id\\": \\"about-this-visualization\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"This page allows you to explore three scenarios for future poverty levels both globally and by region up to 2030. The baseline scenario uses current forecasts for growth in consumption and current inequality levels. The best-case scenario shows what happens with more rapid growth that is shared more equally. By contrast, the worst-case scenario sees slower growth and increasing inequality. Forecasting future poverty levels is difficult, given the inherent uncertainties involved. Poverty modelling develops future scenarios to help inform policymaking. These models rely on simplifying assumptions, often focusing on economic growth and income distribution trends which determine how much poor people benefit from that growth.Data accessed from source: 20 August 2013.\\",
    \\"title\\": \\"About this visualization\\",
  },
  Object {
    \\"id\\": \\"scenario details\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The best and worst case scenarios are based on the extremes of two elements: For growth, a +/- 2 percentage point margin of error either side of baseline projections of consumption growth (which are driven by economic growth). This is in line with differences observed between past forecasts and actual outcomes. For inequality, an annual 0.25 percentage point change in the shares of national consumption between the poorest 40% and the richest 10%\\",
    \\"title\\": \\"Scenario details\\",
  },
  Object {
    \\"id\\": \\"about-the-data\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Projections based on Chandy, Ledlie and Penciakova, The Final Countdown: Prospects for Ending Extreme Poverty by 2030 (2013), using the same methodology applied to a dataset updated with 40 country surveys. All data for years after 2010 are forecasts\\",
    \\"title\\": \\"About the data\\",
  },
]"
`;

exports[`Page data should return page data for a country profile 1`] = `
"Array [
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"introduction\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"Explore this in-depth profile of {country} to find out overall levels of poverty, income distribution, division of wealth and more. Discover how national and sub-national revenue is generated\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"overview-ctry-poorest-people\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MANY OF THE POOREST PEOPLE GLOBALLY LIVE IN {country}?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"overview-ctry-resources\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT RESOURCES ARE AVAILABLE?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"overview-govt-spentperperson\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH DOES THE GOVERNMENT SPEND PER PERSON?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"poverty-reduction-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"IS POVERTY REDUCING OVER TIME?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"poverty-depth-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW DEEP IS POVERTY?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"income-distribution-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW IS INCOME DISTRIBUTED?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"popn-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE POPULATION?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"popn-distribution\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE URBAN VS RURAL SPLIT?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"popn-distribution-age-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE AGE PROFILE?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"totalrevenue-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE TOTAL REVENUE OF {country}?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"govt-revenue-external\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH GOVERNMENT REVENUE COMES FROM EXTERNAL GRANTS (AID)?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"ctry-expenditure\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW IS SPENDING ALLOCATED?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"mixture-of-resources\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHATâ€™S THE MIX OF RESOURCES?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"govt-finance-lower\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"How does {country} finance and spend its budget?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"intl-resources-lower-A1\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"A dynamic mix of resources flow to and from {country}. All resources can play a role in ending poverty, though different resources impact poverty directly and indirectly through different mechanisms and flow for wide-ranging objectives.\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"intl-resources-lower-A2\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"To understand the value of different resources for poverty reduction we must start by understanding the volumes - move the slider to explore how official, commercial, private and illicit resources flow to and from {country}\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"intl-resources-lower-B1\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"Where are international resources originating and where are they destined? Start by selecting inflows or outflows:\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"avg-income-per-person\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE AVERAGE INCOME PER PERSON PER YEAR?\\",
  },
  Object {
    \\"donor_title\\": \\"AS A SHARE OF GNI, HOW MUCH AID DOES {country} ALLOCATE?\\",
    \\"id\\": \\"share-gni-allocated-to-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"AS A SHARE OF GNI, HOW MUCH AID IS ALLOCATED TO {country}?\\",
  },
  Object {
    \\"donor_title\\": \\"HOW HAVE RESOURCE OUTFLOWS CHANGED OVER TIME?\\",
    \\"id\\": \\"resource-inflow\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW HAVE RESOURCE INFLOWS CHANGED OVER TIME?\\",
  },
]"
`;

exports[`Page data should return page data for about 1`] = `
"Array [
  Object {
    \\"id\\": \\"about-the-development-data-hub\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"ABOUT THE DEVELOPMENT DATA HUB\\",
  },
  Object {
    \\"id\\": \\"the-development-data-hub\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The Development Data Hub is the most comprehensive online resource for financial and resource flow data alongside poverty, social and vulnerability indicators. It combines an extensive data store with interactive visualisations enabling you to chart, map and compare data at the global, national and local level.\\",
    \\"title\\": \\"The Development Data Hub\\",
  },
  Object {
    \\"id\\": \\"how-does-it-work?\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"It brings together many data sets,enabling you to dig into these through interactive maps, charts and visualisations. It turns complex data around poverty and resource flows into easy to understand robust information and evidence about what is really going on. The Data Hub is an accessible and easy to use tool for anyone wanting to know more about how resources for development are spent.\\",
    \\"title\\": \\"How does it work?\\",
  },
]"
`;

exports[`Page data should return page data for front page 1`] = `
"Array [
  Object {
    \\"id\\": \\"about-the-development-data-hub\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"ABOUT THE DEVELOPMENT DATA HUB\\",
  },
  Object {
    \\"id\\": \\"the-development-data-hub\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The Development Data Hub is the most comprehensive online resource for financial and resource flow data alongside poverty, social and vulnerability indicators. It combines an extensive data store with interactive visualisations enabling you to chart, map and compare data at the global, national and local level.\\",
    \\"title\\": \\"The Development Data Hub\\",
  },
  Object {
    \\"id\\": \\"how-does-it-work?\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"It brings together many data sets,enabling you to dig into these through interactive maps, charts and visualisations. It turns complex data around poverty and resource flows into easy to understand robust information and evidence about what is really going on. The Data Hub is an accessible and easy to use tool for anyone wanting to know more about how resources for development are spent.\\",
    \\"title\\": \\"How does it work?\\",
  },
]"
`;

exports[`Page data should return page data for front page 2`] = `
"Array [
  Object {
    \\"id\\": \\"comprehensive\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"is the most comprehensive source for financial resource flow data alongside poverty,social and vulnerability indicators.\\",
    \\"title\\": \\"comprehensive\\",
  },
  Object {
    \\"id\\": \\"read\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"Read more about the data hub.\\",
  },
]"
`;

exports[`Page data should return page data for spotlight page 1`] = `
"Array [
  Object {
    \\"id\\": \\"comprehensive-source\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"is a comprehensive source of {country}'s financial resource flow data at the sub-national ({region}) level, alongside indicators on poverty, population, education,health, water, hygiene and sanitation. It highlights the geographical variance in sector performance and financial resources, and seeks to answer whether resources are allocated according to need. Explore the country picture by selecting topics and click on a {region} for an in-depth profile.\\",
    \\"title\\": \\"\\",
  },
]"
`;

exports[`Page data should return page data for the ODA donor bubble chart 1`] = `
"Array [
  Object {
    \\"id\\": \\"diff-providers-priorities\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"How effectively are donors targeting the countries with the highest numbers of people in poverty and the least domestic public resources to address it? This chart allows you to select a provider of aid to see how it allocates aid across countries\\",
    \\"title\\": \\"Different providers, different priorities\\",
  },
]"
`;

exports[`Page data should return page data for the Poverty bubble chart 1`] = `
"Array [
  Object {
    \\"id\\": \\"resources-able-to-meet-needs\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The proportion of people living in extreme poverty worldwide has more than halved since 2000, yet progress is varied across countries. Although domestic public resources are an important resource to drive efforts to end poverty, they remain scarce in many countries where poverty is high\\",
    \\"title\\": \\"Are domestic public resources able to meet the needs of the poorest people?\\",
  },
]"
`;

exports[`Page data should return page data for the Unbundling Oda 1`] = `
"Array [
  Object {
    \\"id\\": \\"diff-providers-priorities\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"How effectively are donors targeting the countries with the highest numbers of people in poverty and the least domestic public resources to address it? This chart allows you to select a provider of aid to see how it allocates aid across countries\\",
    \\"title\\": \\"Different providers, different priorities\\",
  },
]"
`;

exports[`Page data should return page data for unbundling-aid page 1`] = `
"Array [
  Object {
    \\"id\\": \\"explore\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"Explore and compare funding priorities for\\",
  },
  Object {
    \\"id\\": \\"oofs\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The Other Official Flows (OOF) data shown does not include export credits and the quality and completeness of data is reliant on the reporting of OOF activities by individual donors to the CRS.\\",
    \\"title\\": \\"\\",
  },
  Object {
    \\"id\\": \\"click-boxes\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\" Click the boxes to drill down into {aid} bundle. Drag and drop the buttons below to change the order and take different journeys through the data. Use the â€™Compare +â€™ button to see how different countriesâ€™ {aid} bundles compare.\\",
    \\"title\\": \\"\\",
  },
]"
`;
